cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -142,6 +136,60 @@ impl fmt::Display for YaccGrammarErrorKind {
     }
 }
 
+/// Indicates how to interpret the spans of an error.
+pub enum SpansKind {
+    /// The first span is the first occurrence, and a span for each subsequent occurrence.
+    DuplicationError,
+    /// Contains a single span at the site of the error.
+    Error,
+}
+
+impl YaccGrammarError {
+    /// Returns the spans associated with the error, always containing at least 1 span.
+    ///
+    /// Refer to [SpansKind] via [spans_kind](Self::spans_kind)
+    /// for the meaning and interpretation of spans and their ordering.
+    pub fn spans(&self) -> impl Iterator<Item = Span> + '_ {
+        self.spans.iter().copied()
+    }
+
+    /// Returns the kind of spans associated with this error.
+    pub fn spans_kind(&self) -> SpansKind {
```

Comment:
Hmm, should this be `spans_kind` or `spanskind`? I'm not sure!

Reply:
Good catch, normally I would go with `spans_kind`, I could give some other projects that use `foo_kind` in similar situations,
But to be self-consistent there are already other methods, which would imply it should be `spanskind`.

https://docs.rs/lrpar/latest/lrpar/struct.CTParserBuilder.html#method.yacckind
https://docs.rs/lrlex/latest/lrlex/struct.CTLexerBuilder.html#method.lexerkind

Reply:
Fixed in https://github.com/softdevteam/grmtools/pull/322/commits/ed97c1b38b3b2dfd03adae370faf4b2b2f685cff